rbac/api/cross_access/view.py

```diff
@@ -107,6 +116,54 @@ def list(self, request, *args, **kwargs):
             return self.replace_user_id_with_info(result)
         return result
 
+    def partial_update(self, request, *args, **kwargs):
+        """Patch a cross-account request."""
+        validate_uuid(kwargs.get("pk"), "cross-account request uuid validation")
+        payload = request.data
+        for field in payload:
+            if field not in VALID_PATCH_FIELDS:
+                key = "cross-account request"
+                message = f"Field '{field}' is not supported. Please use one or more of: {VALID_PATCH_FIELDS}."
+                error = {key: [_(message)]}
+                raise serializers.ValidationError(error)
+
+        with tenant_context(Tenant.objects.get(schema_name="public")):
+            current = self.get_object()
+            update_data = {
+                "start_date": request.data.get("start_date", current.start_date),
+                "end_date": request.data.get("end_date", current.end_date),
+                "roles": request.data.get("roles", current.roles.all()),
+                "status": request.data.get("status", current.status),
+            }
+            # Todo: This function is pretty messy, and basically just a rehash of the create version,
+            #       can likely clean it up and maybe provide a partial flag
+            self.validate_and_get_input_for_update(update_data)
+
+            if current.status == "expired":
+                update_data["target_account"] = current.target_account
+                if isinstance(update_data.get("roles"), QuerySet):
+                    update_data["roles"] = [role.display_name for role in update_data.get("roles")]
+                request.data.update(update_data)
+                return super().create(request=request, args=args, kwargs=kwargs)
+
+            self.update_cross_account_request(current, update_data)
+            return Response(CrossAccountRequestDetailSerializer(current).data)
+
+    def update(self, request, *args, **kwargs):
+        """Update a cross-account request."""
+        validate_uuid(kwargs.get("pk"), "cross-account request uuid validation")
+
+        with tenant_context(Tenant.objects.get(schema_name="public")):
+            current = self.get_object()
+
+            self.validate_and_get_input_for_creation(request.data)
+
+            if current.status == "expired":
```

Comment:
This would create a new cross-account request but the old one is still there. So when user update it, it still shows up. And it might be better to keep the request_id the same, in the future, we would use that to check the history of the request. Or use super.update? Maybe we could still call the update_cross_account_reuqest, and update the status of it?